doc/usage.md

```diff
@@ -0,0 +1,422 @@
+| END_BY | 'end' | LIKE '%value' | | 
+| IS_NULL | 'null' | IS NULL | set up value is ignored | 
+| IS_NOT_NULL | 'notNull' | IS NOT NULL | set up value is ignored | 
+
+use oat\search\helper\SupportedOperatorHelper to see all supported operators
+
+## usage examples :
+
+### simple value example :
+
+```php
+$query->add(RDFS_LABEL)->equals('foo');
+$query->add(RDFS_LABEL)->gte(1);
+```
+
+OR :
+
+```php
+$query->addCriterion(RDFS_LABEL , SupportedOperatorHelper::EQUAL , 'foo');
+$query)->addCriterion(RDFS_LABEL , SupportedOperatorHelper::GREATER_THAN_EQUAL , 1);
+```
+
+### between value example :
+
+```php
+$query->add(RDFS_LABEL)->between(1 , 10);
+```
+
+OR :
+
+```php
+$query->add(RDFS_LABEL)->between([1 , 10]);
+```
+
+OR :
+
+```php
+$query->addCriterion(RDFS_LABEL,SupportedOperatorHelper::BETWEEN , [1 , 10]);
+```
+
+### IN value example :
+
+```php
+$query->add(RDFS_LABEL)->in(1 , 5 , 10);
+```
+
+OR
+
+```php
+$query->add(RDFS_LABEL)->in([1 , 5 , 10]);
+```
+
+OR
+
+```php
+$query->addCriterion( RDFS_LABEL ,SupportedOperatorHelper::IN , [1 , 5 , 10]);
+```
+
+### NULL and NOT NULL value example :
+
+```php
+$query->add(RDFS_LABEL)->null(NULL);
+$query->add(RDFS_LABEL)->notNull(NULL);
+```
+
+OR
+
+```php
+$query->addCriterion( RDFS_LABEL ,SupportedOperatorHelper::NULL , null);
+$query->addCriterion( RDFS_LABEL ,SupportedOperatorHelper::NOT_NULL , null]);
+```
+
+## sorting Query :
+
+### Sort method is available on QueryBuilder :
+
+```php
+$search = $this->getServiceManager()->get(\oat\oatbox\search\ComplexSearchService::SERVICE_ID);
+$queryBuilder = $search->query()->sort(
+[
+RDFS_LABEL => 'asc'
+]
+);
+
+$query = $queryBuilder->add(RDFS_COMMENT)->contains('foo');
+$queryBuilder->setCriteria($query);
+
+$result = $search->getGateway()->search($queryBuilder);
+```
+
+### Example for muliple sort :
+
+```php
+$search = $this->getServiceManager()->get(\oat\oatbox\search\ComplexSearchService::SERVICE_ID);
+
+$queryBuilder = $search->query();
+
+$query = $queryBuilder->add(RDFS_COMMENT)->contains('foo');
+
+$queryBuilder->setCriteria($query)->sort(
+[
+RDFS_LABEL => 'asc',
+RDFS_COMMENT => 'desc',
+]
+);
+$result = $search->getGateway()->search($queryBuilder);
+```
+
+### random sorting
+
+```php
+$search = $this->getServiceManager()->get(\oat\oatbox\search\ComplexSearchService::SERVICE_ID);
+
+$queryBuilder = $search->query();
+
+$query = $queryBuilder->add(RDFS_COMMENT)->contains('foo');
+
+$queryBuilder->setCriteria($query)->setRandom();
+
+//fields are ignore when random is enable
+
+$result = $search->getGateway()->search($queryBuilder);
+```
+
+## Limit and Offset :
+
+limit method is also available on QueryBuilder :
+
+get 10 results :
+
+```php
+$search = $this->getServiceManager()->get(\oat\oatbox\search\ComplexSearchService::SERVICE_ID);
+
+$query = $queryBuilder->newQuery()->add(RDFS_COMMENT)->contains('foo');
+
+$queryBuilder->setCriteria($query);
+
+$result = $search->getGateway()->search($queryBuilder)->setLimit(10);
```

Comment:
This breaks (PHP Fatal error:  Call to undefined method oat\generis\model\kernel\persistence\smoothsql\search\TaoResultSet::setLimit())

I think that the setLimit should be called on the queryBuilder.